populus/assets/defaults.v8.config.json

```diff
@@ -186,14 +192,14 @@
       "SolcStandardJSON": {
         "class": "populus.compilation.backends.SolcStandardJSONBackend",
         "settings": {
-          "optimize": true,
           "output_values": [
             "abi",
             "bin",
             "bin-runtime",
             "metadata"
           ],
           "stdin": {
+            "evmVersion": "homestead",
```

Comment:
Honestly, I'm not optimistic about this change (here and earlier in the file).

This forces `solc` to compile with "Homestead rules" (i.e. not using the features introduced in Byzantium, e.g. the `RETURNDATASIZE` operation).

Perhaps it is OK for now, especially since no one else is showing initiative to keep up-to-date; but wouldn't it be clearer to notify the user that "with this shiny new compiler, your code needs a correction"?.. ("Notify the user" by letting the compiler error fall through.)

-----

In either case, this might need a version bump to the config file name (`v8`->`v9`).

Reply:
This will likely be released at the same time as #466 IIUC, which means it would come with a major version bump. So backwards compatibility isn't critical, if `homestead` was there for compatibility.

Reply:
Mirroring my comment from gitter here:
the main problem here is that, `populus` uses `eth-testrpc` for testing. `eth-testrpc` uses `pyethereum` internally. `pyethereum` doesn’t have `returndatasize` and `returndatacopy` implemented which gives a `TransactionFailed` error with versions of solidity >=0.4.22. In fact, I had to add the 2nd commit just to address the test failures in the PR.

Till populus is ported to work with `eth-tester`, we’ll have to stick to the `homestead` workaround!

Reply:
> Notify the user" by letting the compiler error fall through

That's something I'd like to see happen too. But unfortunately `eth-testrpc` is a blocker here. Let me try to port `populus` to newer `web3` and `eth-tester` over the weekend.

Reply:
Ah, understood! IMO the PR is good as-is then: our current toolchain/dependencies only allow `homestead` anyway!